app/utils/ajax-promise.js

```diff
@@ -4,7 +4,8 @@ import config from "../config/environment";
 let defaultHeaders = {
   "X-GOODCITY-APP-NAME": config.APP.NAME,
   "X-GOODCITY-APP-VERSION": config.APP.VERSION,
-  "X-GOODCITY-APP-SHA": config.APP.SHA
+  "X-GOODCITY-APP-SHA": config.APP.SHA,
+  "Accept-Language": JSON.parse(localStorage.language)
```

Comment:
From a general engineering perspective, this is a code smell

- it makes an assumption on how language works in the app 
- it doesn't support change
- it may break (parsing error)

A better solution would be to make the default headers dynamically configurable.

If you scroll down on this page you will notice there's a function `setDefaultHeaders`, which takes in a **FUNCTION**

Notice that the stock app is already using this here: https://github.com/crossroads/stock.goodcity/blob/master/app/instance-initializers/ajax.js

so you can just add the language to the adapter's settings there.

Reply:
@swatijadhav ☝️

Reply:
I would cancel this PR altogether

Reply:
Thanks for the details @patrixr, updated PR